pkg/api/v1/container_test.go

```diff
@@ -0,0 +1,65 @@
+package apiv1
+
+import (
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/beam-cloud/beta9/pkg/auth"
+	"github.com/beam-cloud/beta9/pkg/common"
+	"github.com/beam-cloud/beta9/pkg/repository"
+	"github.com/beam-cloud/beta9/pkg/types"
+	"github.com/labstack/echo/v4"
+	"github.com/tj/assert"
+)
+
+type MockContainerTestDetails struct {
+	backendRepo   *repository.BackendRepository
+	containerRepo *repository.ContainerRepository
+	server        *echo.Echo
+	httpDetails   *repository.MockHttpDetails
+}
+
+func setupContainerEchoGroup() MockContainerTestDetails {
+	configManager, err := common.NewConfigManager[types.AppConfig]()
+	if err != nil {
+		log.Fatalf("Failed to create config manager: %v", err)
+	}
+
+	config := configManager.GetConfig()
+	md := repository.MockBackendWithValidToken()
+
+	rdb, err := repository.NewRedisClientForTest()
+	if err != nil {
+		log.Fatalf("Failed to create redis client: %v", err)
+	}
+
+	containerRepo := repository.NewContainerRedisRepositoryForTest(rdb)
+
+	e := echo.New()
+	e.Use(auth.AuthMiddleware(md.BackendRepo))
+	NewContainerGroup(e.Group("/container"), md.BackendRepo, containerRepo, config)
+
+	return MockContainerTestDetails{
+		backendRepo:   &md.BackendRepo,
+		containerRepo: &containerRepo,
+		server:        e,
+		httpDetails:   &md,
+	}
+}
+
+func TestListContainersByWorkspaceId(t *testing.T) {
+	testDetails := setupContainerEchoGroup()
+
+	// 1. Test with valid workspaceId
+	testDetails.httpDetails.AddExpectedDBTokenSelect(testDetails.httpDetails.Mock, *testDetails.httpDetails.TokenForTest.Workspace, testDetails.httpDetails.TokenForTest)
```

Comment:
What exactly is happening here? This test is a little hard to read - how is this validating that the containers are only containers within your workspace? Or is it just checking that the token is valid?

If it's just checking that the token is valid, we probably don't need a test case for every set of endpoints just to check that

Reply:
I think `MockContainerTestDetails` and `setupContainerEchoGroup` are analogous to  `ContainerGroup` and `NewContainerGroup`. Maybe we should name them that with the the suffix `ForTest` like we've done before?

In regards to the body of the function, we might want to turn the majority of this into a helper function. I can see this being used in most endpoints (with the exception endpoints that won't require auth).

Reply:
This `testDetails.httpDetails.AddExpectedDBTokenSelect` is used to make sure the next call to the db returns the token. Unlike pytest, we don't have fixture data so we have expect db calls and return the exact thing we want through some sort of sqlmock queue.

Reply:
> I think `MockContainerTestDetails` and `setupContainerEchoGroup` are analogous to `ContainerGroup` and `NewContainerGroup`. Maybe we should name them that with the the suffix `ForTest` like we've done before?
> 
> In regards to the body of the function, we might want to turn the majority of this into a helper function. I can see this being used in most endpoints (with the exception endpoints that won't require auth).

I'll add the `ForTest` suffix